### DISCUSSION TITLE
S2 wake on external pin ?

### DISCUSSION THREAD

**Discussion by GrosMoi** [micropython/micropython#13374] [ESP32] [answered]
Hi,

 I'm using an esp32-S2 (S2 mini) and I can't wake it up with an external signal (low level or high level).  The code works fine on an ESP32-WROOM-32D but not on the S2. I'm using the latest version 1.22.1 (LOLIN_S2_MINI-20240105-v1.22.1.bin). I've tried other versions but it's the same. I've also tried other pins. 

With the WAKEUP_ANY_HIGH option, ESP never wakes up

wake = Pin(3,Pin.IN)
esp32.wake_on_ext0(pin = wake, level = esp32.WAKEUP_ANY_HIGH)
machine.deepsleep()    

and with the WAKEUP_ALL_LOW option, ESP wakes up immediately after the deepsleep call.

Is this a bug or is it not possible to do this type of wake-up with S2?
Thanks in advance.

This message is an opportunity to thank all those who worked on this brilliant micropython project!

**Comment by shariltumin** [comment, marked as answer]
I did this on wemos s2 mini

```py
>>> from machine import Pin, deepsleep
>>> import esp32
>>> wake = Pin(21, Pin.IN, Pin.PULL_DOWN)
>>> esp32.wake_on_ext0(pin = wake, level = esp32.WAKEUP_ANY_HIGH)
>>> deepsleep()

[tio 00:07:29] Disconnected
Touch GPIO21 to 3V3
[tio 00:10:04] Connected
```

According to [espressif](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/peripherals/gpio.html) there are only two GPIOs dedicated to RTC_GPIO: RTC_GPIO0, RTC_GPIO21.

GPIO1 to GPIO20 are also used for ADC1 and ADC2, and do not seem to work for the wake pin. Therefore, GPIO21 is the only viable GPIO.

**Comment by GrosMoi** [comment]
hello,

 Thank you very much, it works very well with with GPIO21. It's a bit silly, this hardware limitation on the S2.
